### PR TITLE
Automatically logs user into the charmstore in GISF

### LINKED ISSUE #2446: In GISF you should automatically be logged in to the charmstore
When you are automatically logged in to the controller in gisf mode, we should also log you into the charmstore, or else automatically log you in before you see your profile or entities in the store.

### PR DESCRIPTION
**Pull request by jaycee**
Fixes #2446

### CODE REVIEW

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -1077,8 +1080,12 @@ YUI.add('juju-gui', function(Y) {
       // If the username does not match the logged in user then display a new
       // model instead of the profile.
       const auth = this._getAuth();
+      console.log(JSON.stringify(auth));
+      console.log(auth.rootUserName);
+      console.log(state.profile);
+      console.log(JSON.stringify(auth));
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: OOps :)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6795/

File: jujugui/static/gui/src/app/app.js (hunk 1 above)
Comment (jaycee) [inline jujugui/static/gui/src/app/app.js]: :smile:

**Comment by jaycee** [comment]
QA:

Use guiproxy -config 'gisf:true, jujuEnvUUID:""'
Go /login; you should automatically be logged in and see your profile. You should not see the "log in to charmstore" button. If you have charms or bundles in your username space, you should see them on your profile page.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6814/

File: jujugui/static/gui/src/app/app.js [hunk 2]
```diff
@@ -1078,7 +1081,7 @@ YUI.add('juju-gui', function(Y) {
       // model instead of the profile.
       const auth = this._getAuth();
       if (auth && state.profile !== auth.rootUserName) {
-        this.state.changeState({new: '', profile: null});
+        this.state.changeState({root: 'new', profile: null});
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: Thanks for this driveby

Comment (jaycee) [inline jujugui/static/gui/src/app/app.js]: Thanks for helping me fix it. :smile:

**Comment by bac** [comment]
QA-OK

User and charmstore not getting passed, or some such.  JC knows what this means.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6815/

**Comment by jaycee** [comment]
Addressed the one issue found in @bac's qa.

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6816/
